GH-556: Bootstrap deployment in Eve

gdemonet opened the pull request:
Using results from the build step, we mount the ISO and execute the bootstrap script on a fresh OpenStack VM worker (using CentOS 7).

Fixes GH-556.

gdemonet committed:
Add single-node deployment step

We use an OpenStack worker so we can use the worker itself as target of
the deployment.

Issue: GH-556

gdemonet committed:
Use short version obtained from build

Instead of using `get_product_version.sh` to provide the necessary build
property, we retrieve the information among the built artifacts.

Issue: GH-556

gdemonet committed:
Set property in parent stage

We attempted to set the property in the `build` stage, but doing so was
not sufficient to make its value available in the subsequent
`single-node` stage.

Issue: GH-556

gdemonet committed:
Add mountpoint creation

Issue: GH-556

gdemonet committed:
Run bootstrap script as root

Issue: GH-556

bert-e commented:
# Hello gdemonet,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux

charlesprost commented on eve/main.yml:
```diff
@@ -10,18 +10,28 @@ stages:
-          stage_names:
-            - build
-            - lint
+    - TriggerStages:
```
Is this new indentation yaml-valid?

jbertran replied:
It's actually more valid than before - I didn't include Eve yaml in the yaml linting step since it uses some form of templating that's not directly YAML-valid, but the yaml linter _does_ yell on indentation quite a lot on this content taken from other `main.yml`s. Ditto with pod manifests.

gdemonet replied:
I ran `yamllint` to fix the indentation of the whole file (see 749652c)

gdemonet committed:
Use more reasonable VM flavor for now

Issue: GH-556

gdemonet committed:
Display pretty output from bootstrap execution

Issue: GH-556

gdemonet committed:
Fix indentation and other yamllint errors

Issue: GH-556

bert-e commented:
# Conflict




There is a conflict between your branch `feature/556-bootstrap-deployment` and the
destination branch `development/2.0`.

Please resolve the conflict on **the feature branch** (`feature/556-bootstrap-deployment`).

```bash
 $ git fetch
 $ git checkout origin/feature/556-bootstrap-deployment
 $ git merge origin/development/2.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/556-bootstrap-deployment
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux

gdemonet committed:
Use generic pattern to get salt-master Pod

Issue: GH-556

gdemonet committed:
Halt CI run if either lint or build fails

It could be interesting to split lint and build steps, in order to
avoid the long (roughly 10 minutes at the time of writing) build step
if the lint would fail anyway.

Issue: GH-556

bert-e commented:
# Conflict




There is a conflict between your branch `feature/556-bootstrap-deployment` and the
destination branch `development/2.0`.

Please resolve the conflict on **the feature branch** (`feature/556-bootstrap-deployment`).

```bash
 $ git fetch
 $ git checkout origin/feature/556-bootstrap-deployment
 $ git merge origin/development/2.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/556-bootstrap-deployment
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux

gdemonet committed:
Do not use a pseudo-terminal for running bootstrap

It breaks in strange ways (e.g. `systemctl start containerd` hangs and
times out). Let's forget about colors in the CI.

Issue: GH-556

gdemonet commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux










*The following options are set:* **approve**

gdemonet commented:
@bert-e wait

gdemonet commented:
> @bert-e wait

This was set so we can warn other PR authors that deployment is now part of the CI and needs to pass (bootstrap script return code 0) to get a green build.

jbertran reviewed (approved):
LGTM. Let's add 20 minutes to everyone's build :upside_down_face:

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye gdemonet.
